app-modules/assistant/src/Filament/Pages/PersonalAssistant.php

```diff
@@ -220,9 +221,14 @@ public function sendMessage(): void
     public function ask(AIChatClient $ai): void
     {
         try {
-            $this->currentResponse = $ai->ask($this->chat, function (string $partial) {
-                $this->stream('currentResponse', nl2br($partial));
-            });
+            /** @var User $user */
+            $user = auth()->user();
+
+            $this->currentResponse = $ai
+                ->provideDynamicContext(new DynamicContext($user))
+                ->ask($this->chat, function (string $partial) {
+                    $this->stream('currentResponse', nl2br($partial));
+                });
```

Comment:
@dgoetzit and @mxm1070 the way this works does it include it as a system message once. Or is it included with ever prompt that is sent by the User?

Reply:
I believe when I tested it it was always appended.

Reply:
@mxm1070 got it. Thank you!

@dgoetzit and @danharrin with y'all experience with the AI do you think we should do it this way? I am thinking not, as it increases the tokens per prompt doesn't it? And could result in strange messages.

Perhaps we want a way to have dynamic context included at the start of a conversation. Then every once an a while after.

We don't have to decide or change this now. Just want to discuss it for future items.

Reply:
This context should be included in every message.

Reply:
@Orrison In theory that makes a ton of sense, but I'm not sure how that would impact the response. I don't recall if the system context is preserved over multiple requests. But, if it is (even just for _x_ period of time), and if we could determine exactly how often we'd need to provide that context again without negatively impacting the results I'd be in favor of it, as it would reduce usage even if it is only a relatively small token savings per conversation.